ignite/contrib/metrics/regression/geometric_mean_relative_absolute_error.py

```diff
@@ -34,23 +36,47 @@ class GeometricMeanRelativeAbsoluteError(_BaseRegression):
             non-blocking. By default, CPU.
     """
 
+    @reinit__is_reduced
     def reset(self) -> None:
-        self._sum_y = 0.0  # type: Union[float, torch.Tensor]
-        self._num_examples = 0
-        self._sum_of_errors = 0.0  # type: Union[float, torch.Tensor]
+        self._predictions = []  # type: List[torch.Tensor]
+        self._targets = []  # type: List[torch.Tensor]
 
     def _update(self, output: Tuple[torch.Tensor, torch.Tensor]) -> None:
-        y_pred, y = output
-        self._sum_y += y.sum()
-        self._num_examples += y.shape[0]
-        y_mean = self._sum_y / self._num_examples
-        numerator = torch.abs(y.view_as(y_pred) - y_pred)
-        denominator = torch.abs(y.view_as(y_pred) - y_mean)
-        self._sum_of_errors += torch.log(numerator / denominator).sum()
+        y_pred, y = output[0].detach(), output[1].detach()
+
+        y_pred = y_pred.clone().to(self._device)
+        y = y.clone().to(self._device)
+
+        self._predictions.append(y_pred)
+        self._targets.append(y)
 
     def compute(self) -> float:
-        if self._num_examples == 0:
+        if len(self._predictions) < 1 or len(self._targets) < 1:
             raise NotComputableError(
                 "GeometricMeanRelativeAbsoluteError must have at least one example before it can be computed."
             )
-        return torch.exp(torch.mean(cast(torch.Tensor, self._sum_of_errors) / self._num_examples)).item()
+
+        _prediction_tensor = torch.cat(self._predictions, dim=0)
+        _target_tensor = torch.cat(self._targets, dim=0)
+
+        ws = idist.get_world_size()
+
+        if ws > 1:
```

Comment:
I was wondering if all procs have the tensors, why let only proc 0 does the computation and add an extra communication. Why not duplicate the calculation over all procs and avoid the communication L80 ?

Reply:
I see your point, I don't actually what would be better between these two implementations, but my point here I wanted to keep the implementation similar to something that we already have (as almost the rest of the metric here has the same implementation), I wanted this to be almost the same as `EpochMetric`, I don't know what would be better here, please tell me what do you think?

Reply:
In case of EpochMetric we execute a custom function on cpu (think of a sklearn metric) and it doesn't make much sense to execute this function on all processes (especially if there are any parallelism in the implementation).

Reply:
Thanks for explanation. OK, I will change the implementation

Reply:
There could have been a compromise between communication and calculation. Anyway, it was a simple remark.

Reply:
Thanks @sdesrozis for pointing that out!

Reply:
> Thanks for explanation. OK, I will change the implementation